app/src/main/java/com/sdp13epfl2021/projmag/activities/ProfilePageActivity.kt

```diff
@@ -30,23 +43,73 @@ class ProfilePageActivity : AppCompatActivity() {
             val gallery = Intent(Intent.ACTION_PICK, MediaStore.Images.Media.INTERNAL_CONTENT_URI)
             startActivityForResult(gallery, pickImage)
         }
+
+
+        UserProfileDatabase(Firebase.firestore, Firebase.auth).getProfile(::loadUserProfile)
+
         buttonAddCv.setOnClickListener{
             val intent = Intent(this,CVCreationActivity::class.java)
             startActivity(intent)
             finish()
         }
         buttonSubChange.setOnClickListener{
+            val profile = createProfileFromFields()
+            if(profile != null) {
+                UserProfileDatabase(Firebase.firestore, Firebase.auth).uploadProfile(profile, {}, {})
+            }
```

Comment:
Could be rewrite as
```kotlin
createProfileFromFields()?.{ profile -> UserProfileDatabase(..) }
```
However it is subjective, it just remove one line of code

Reply:
I tried to write it like that but it did not work maybe I am doing something wrong?
Will keep it like that in the mean time.